astroid/nodes/node_ng.py

```diff
@@ -248,15 +263,40 @@ def parent_of(self, node):
                 return True
         return False
 
-    def statement(self):
+    @overload
+    def statement(
+        self, future: Literal[None] = ...
+    ) -> Union["nodes.Statement", "nodes.Module"]:
+        ...
+
+    @overload
+    def statement(self, future: Literal[True]) -> "nodes.Statement":
+        ...
+
+    def statement(
+        self, future: Literal[None, True] = None
+    ) -> Union["nodes.Statement", "nodes.Module", NoReturn]:
         """The first parent node, including self, marked as statement node.
 
-        :returns: The first parent statement.
-        :rtype: NodeNG
+        TODO: Deprecate the future parameter and only raise StatementMissing and return
+        nodes.Statement
+
+        :raises AttributeError: If self has no parent attribute
+        :raises StatementMissing: If self has no parent attribute and future is True
         """
         if self.is_statement:
-            return self
-        return self.parent.statement()
+            return cast("nodes.Statement", self)
+        if not self.parent:
+            if future:
+                raise StatementMissing(target=self)
+            warnings.warn(
+                "In astroid 3.0.0 NodeNG.statement() will return either a nodes.Statement "
+                "or raise a StatementMissing exception. This behaviour can already be triggered "
+                "by passing 'future=True' to a statement() call.",
```

Comment:
I added something about `AttributeError` but think it might be helpful to fully explain what `statement()` will now do. Saves the maintainer/contributor changing the code some time!

Reply:
👌🏻